File: C PROGRAMS/armstrong.c
#include<stdio.h>
#include<math.h>

void main(){

    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    //sum of digits 
    int temp = num;
    int sum = 0;
    while(temp > 0){
        sum = sum + (temp%10);
        temp = temp / 10;
    }

    int reverse_sum = 0;
    temp = sum;
    while(temp > 0){
        int d = temp%10;
        reverse_sum = reverse_sum * 10 + d;
        temp = temp/10;
    }

    if((sum *  reverse_sum) == num)printf("Magic number");
    else printf("Not a magic number");


}

File: JAVA DSA SHEET/Know_Basic_Maths/Armstrong.java
package Know_Basic_Maths;
import java.util.Scanner;

public class Armstrong {
    public static void main(String[] args) {

        Scanner sc=new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = sc.nextInt();
        sc.close();

        int temp = num;

        int count = 0;
        while(temp>0){
            count++;
            temp/=10;
        }

        temp = num;
        int result=0;

        while(temp>0){
            int d = temp%10;
            result = result + (int) Math.pow(d, count);
            temp/=10;
        }

        if(result == num)System.out.println("Armstrong number");
        else System.out.println("Not armstrong number");
        
    }
}
